XiEditor/AppDelegate.swift

```diff
@@ -107,147 +99,96 @@ class AppDelegate: NSObject, NSApplicationDelegate {
         dispatcher.coreConnection.sendRpcAsync(req.method, params: req.params!)
     }
 
-    /// returns the NSDocument corresponding to the given viewIdentifier
-    private func documentForViewIdentifier(viewIdentifier: ViewIdentifier) -> Document? {
-        for doc in NSApplication.shared.orderedDocuments {
-            guard let doc = doc as? Document else { continue }
-            if doc.coreViewIdentifier == viewIdentifier {
-                return doc
-            }
-        }
-        return nil
+    // MARK: XiClient protocol
```

Comment:
We've been using the `// MARK: -` syntax, which I think is appropriate here, but that's the tiniest of nits.

Reply:
Good nit.

An annoyance of mine is that `"//MARK: - "` as a normal comment adds an underline + a heading to the (little-used I suppose) file navigator in Xcode; but in doc comments for special sections you need the dash first, e.g. `"/// - Note: This function blah blah blah"`, in order for the inline documentation to render properly.

(after some playing around) actually after playing around: apparently you can have the dash before _or_ after the mark, and get a line break in the navigator:

![screen shot 2018-01-04 at 6 21 47 pm](https://user-images.githubusercontent.com/3330916/34588676-467bc99a-f17c-11e7-83da-094ab3e77d60.png)

(apologies if everyone knows this but me)